commercial-paper/README.md

```diff
@@ -221,12 +514,22 @@ npm install
 
 The docker containers don't contain the node or Java runtimes; so it is best to exit the docker containers - but keep the windows open and run the applications locally.
 
-### Issue the paper
+As mentioned earlier in the Sample introduction section, transaction _inputs_ are recorded on the ledger, as well as any asset _state_ changes. Just *before* you run the _issue_ application script below - you need to launch a block 'listener' application that will show you these _inputs_, as you complete each transaction in the commercial paper lifecycle (eg. Paper Number: 00001, 00002 etc) . 
 
-This is running as *MagnetoCorp* These commands are to be run in the
-`commercial-paper/organization/magnetocorp/application` directory or the `commercial-paper/organization/magnetocorp/application-java`
+For the listener, its best to open a *new* terminal for this in the `commercial-paper/organization/magnetocorp/application` directory (javascript). Next, run the `addToWallet` step in the `issue` transaction below, to add Isabella's identity to the wallet - the listener will use this wallet. Once the listener is launched, it will show the inputs for transactions you will perform and which are committed to blocks (ie part of the ledger).  Note: initially, the listener may show a spurious message, and then go into a _listening_ or 'wait' state. As transactions complete below, messages will be displayed by the listener - so keep an eye out. *After* adding Isabella's wallet, you can then launch the listener as follows:
 
-*Add the Identity to be used*
+```
+node cpListener.js
+```
```

Comment:
I find this section quite confusing. Here is a suggestion for simplification:

For the listener, it is best to open a *new* terminal in which the listener will show the inputs for the transactions you perform and which are committed to blocks (ie part of the ledger).

Because the listener will use Isabella's identity from the same wallet it needs to be launched **after** the `addToWallet` step in the `issue` transaction below. You can then launch the listener from your new terminal as follows:

```
cd fabric-samples/commercial-paper/organization/magnetocorp/application
node cpListener.js
```

**Note**: initially, the listener may show a spurious message, and then go into a _listening_ or 'wait' state. As more transactions complete, messages will be displayed by the listener - so keep an eye out.